src/Swarm/Util.hs

```diff
@@ -349,7 +351,13 @@ _NonEmpty :: Lens' (NonEmpty a) (a, [a])
 _NonEmpty = lens (\(x :| xs) -> (x, xs)) (const (uncurry (:|)))
 
 ------------------------------------------------------------
--- Some utilities for NP-hard approximation
+-- Some set utilities
+
+-- | Remove any sets which are supersets of other sets.  In other words,
+--   (1) no two sets in the output are in a subset relationship
+--   (2) every element in the input is a superset of some element in the output.
+removeSupersets :: Ord a => Set (Set a) -> Set (Set a)
+removeSupersets ss = S.filter (\s -> all (liftA2 (||) (== s) (not . (`S.isSubsetOf` s))) ss) ss
```

Comment:
Would some intermediate variables improve comprehensibility?

Reply:
I've refactored it a lot and I think it's much more readable now:
```
removeSupersets ss = S.filter (not . isSuperset) ss
 where
  isSuperset s = any (`S.isSubsetOf` s) (S.delete s ss)
```

Reply:
Very nice!